src/components/Calculator.js

```diff
@@ -1,42 +1,76 @@
 import React, { Component } from "react";
+import calculate from "../logic/calculate";
+import operate from "../logic/operate";
 
 export class Calculator extends Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {calcoutput: 0,calcobj:{},calcnext:null,calctotal:null};
+  }
+  
   render() {
+    const clickbutton = (inp) => (event) => {     
+      //console.log(calculate({},this.state.calcobj))
+      let returnobj = calculate(this.state.calcobj,inp);
+      this.state.calcobj = returnobj;
```

Comment:
- [ ] You should not mutate state directly. Use setState()  instead

Reply:
Changed my code to use setState() in all cases. I noticed mutating it directly updates the state but doesn't rerender the component, I thought this was usefull, will always use setState() from now on. Thanks for pointing that out.